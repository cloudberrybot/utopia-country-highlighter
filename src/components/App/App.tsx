import { Header } from '../Header';
import { Navbar } from '../Navbar';
import React from 'react';
import { Regions } from '../Regions';
import { useGetCountriesQuery } from '../../services/countries/apiSlice';

function App() {
  const { data, error, isLoading } = useGetCountriesQuery();
  console.log({ data, error, isLoading });

  return (
    <div>
      <Navbar />
      <div className="container mx-auto">
        <Header />
        <Regions />
      </div>
    </div>
  );
}

export default App;
