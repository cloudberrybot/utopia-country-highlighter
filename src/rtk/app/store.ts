import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit';

import { apiSlice } from '../services/countries/apiSlice'
import countriesSlice from '../features/countries/countriesSlice';

export const store = configureStore({
  reducer: {
    [apiSlice.reducerPath]: apiSlice.reducer,
    countries: countriesSlice,
  },
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware().concat(apiSlice.middleware)
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
