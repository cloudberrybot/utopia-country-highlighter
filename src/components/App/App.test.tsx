import App from './App';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { store } from '../../rtk/app/store';

test('renders correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
