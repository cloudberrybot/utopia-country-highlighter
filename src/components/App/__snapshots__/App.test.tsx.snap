// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="overflow-x-hidden"
  >
    <div
      class="grid grid-cols-3 md:grid-cols-4 gap-4 h-[100px] md:h-[183px] bg-navbar items-center"
    >
      <div
        class="col-span-2 md:col-start-2 md:col-end-4 md:justify-self-center"
      >
        <div
          class="text-lg md:text-4xl lg:text-5xl text-white font-sans pl-6 min-w-full whitespace-nowrap"
        >
          Utopia Country Highlighter
        </div>
      </div>
      <div
        class="relative w-14 h-14 md:w-24 md:h-24 mr-2 md:mr-6 justify-self-end"
      >
        <img
          alt="logo"
          class="rounded-full"
          src="logo.png"
        />
        <span
          class="flex absolute h-5 w-5 top-0 md:top-2 right-1 md:right-2 -mt-1 -mr-1"
        >
          <span
            class="relative inline-flex rounded-full h-4 w-4 bg-green-500"
          />
        </span>
      </div>
    </div>
    <div
      class="container mx-auto"
    >
      <div
        class="mt-8 ml-2 mb-2"
      >
        <h1
          class="text-lg text-gray-700 font-semibold"
        >
          Choose a Continent
        </h1>
        <hr
          class="border-b border-gray-200 border-[1px] mt-1"
        />
      </div>
      <div
        class="grid grid-cols-3 justify-items-stretch md:grid-cols-4 lg:grid-cols-6 gap-4"
      >
        <button
          class="text-gray-800 text-lg md:text-xl"
          type="button"
        >
          Africa
        </button>
        <button
          class="text-gray-800 text-lg md:text-xl"
          type="button"
        >
          Americas
        </button>
        <button
          class="text-gray-800 text-lg md:text-xl"
          type="button"
        >
          Asia
        </button>
        <button
          class="text-gray-800 text-lg md:text-xl"
          type="button"
        >
          Europe
        </button>
        <button
          class="text-gray-800 text-lg md:text-xl"
          type="button"
        >
          Oceania
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
